### PR TITLE
Add DeleteNamespace API to operator service

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add DeleteNamespace API to operator service.

**Why?**
Expose delete namespace feature to frontend.

**How did you test it?**
Implemented on server.

**Potential risks**
No risks.

### CODE REVIEW

File: temporal/api/operatorservice/v1/service.proto [hunk 1]
```diff
@@ -54,7 +54,15 @@ service OperatorService {
     rpc RemoveSearchAttributes (RemoveSearchAttributesRequest) returns (RemoveSearchAttributesResponse) {
     }
 
-    // GetSearchAttributes returns comprehensive information about search attributes.
+    // ListSearchAttributes returns comprehensive information about search attributes.
     rpc ListSearchAttributes (ListSearchAttributesRequest) returns (ListSearchAttributesResponse) {
     }
+
+    // DeleteNamespace synchronously deletes namespace and asynchronously reclaim all namespace resources.
```
Comment (Sushisource) [inline temporal/api/operatorservice/v1/service.proto:61]:
```suggestion
    // DeleteNamespace synchronously deletes a namespace and asynchronously reclaims all namespace resources.
```

File: temporal/api/operatorservice/v1/request_response.proto [hunk 2]
```diff
@@ -61,4 +61,22 @@ message ListSearchAttributesResponse {
...
+//     aip.dev/not-precedent: DeleteNamespace RPC doesn't follow Google API format. --)
+// (-- api-linter: core::0135::request-name-required=disabled
+//     aip.dev/not-precedent: DeleteNamespace RPC doesn't follow Google API format. --)
+message DeleteNamespaceRequest {
+    string namespace = 1;
+    string namespace_id = 2;
+    int32 delete_activity_rps = 3;
+    int32 page_size = 4;
+    int32 pages_per_execution_count = 5;
+    int32 concurrent_delete_executions_activities = 6;
```
Comment (Sushisource) [inline temporal/api/operatorservice/v1/request_response.proto:76]: Some comments on these would be welcome. It's not immediately obvious to me what any of them are for

Comment (alexshtin) [reply temporal/api/operatorservice/v1/request_response.proto:76]: I agree, after you made such a great job adding comments to almost every field, it is shame to add request/response objects w/o comments!

File: temporal/api/operatorservice/v1/request_response.proto [hunk 3]
```diff
@@ -59,6 +59,38 @@ message ListSearchAttributesResponse {
...
+}
+
+// (-- api-linter: core::0135::request-unknown-fields=disabled
+//     aip.dev/not-precedent: DeleteNamespace RPC doesn't follow Google API format. --)
+// (-- api-linter: core::0135::request-name-required=disabled
+//     aip.dev/not-precedent: DeleteNamespace RPC doesn't follow Google API format. --)
+message DeleteNamespaceRequest {
+    // One of namespace or namespace_id must be specified to identify namespace.
+    string namespace = 1;
+    string namespace_id = 2;
```
Comment (Sushisource) [inline temporal/api/operatorservice/v1/request_response.proto:73]: Should this be a oneof then? If both are specified, will the request be rejected?

File: temporal/api/operatorservice/v1/request_response.proto [hunk 4]
```diff
@@ -59,6 +59,38 @@ message ListSearchAttributesResponse {
...
+
+    // 4 parameters to fine configuration of delete executions workflow. Use with caution.
```
Comment (Sushisource) [inline temporal/api/operatorservice/v1/request_response.proto:75]:
```suggestion
    // 4 parameters to configure the delete executions workflow. Use with caution.
```

**Review by Sushisource — APPROVED** [review]
Awesome, thanks for the comments explainers

File: temporal/api/operatorservice/v1/request_response.proto (hunk 3 above)
Comment (alexshtin) [reply temporal/api/operatorservice/v1/request_response.proto]: At the end of the day I decided to simplify it significantly.

**Review by yiminc — APPROVED** [review]

**Pull request merged**